lib/src/download_utils.dart

```diff
@@ -43,7 +43,12 @@ Future<void> downloadPackage(
       try {
         final req = await c.getUrl(packageUri);
         final res = await req.close();
+        if (res.statusCode >= 500) {
+          // may be a temporary failure, let's retry...
+          throw Exception('Unable to access URL: "$packageUri".');
+        }
         if (res.statusCode != 200) {
+          // throwing error stops retrying (e.g. on 404)
           throw AssertionError('Unable to access URL: "$packageUri".');
```

Comment:
I think we should still find something else to throw here. AssertionError is not really meant for this purpose. This might require making a custom exception class and a passing a `retryIf` parameter.

Reply:
As long as it is an `Error`, we don't need to pass it to `retryIf`.

Reply:
My point is: it is not right to throw an error here. Errors are for unrecoverable situations, and failing to download a file should be recoverable.

AssertionError has a special meaning, it is thrown when an Assertion fails - I would never expect that in production (that is running without asserts).

Reply:
Maybe we should just depend on `package:http_retry` (which we already do in pub-dev), for such logic?